### PR TITLE
Add practice exercise `rectangles`

### PR DESCRIPTION
**Pull request by jiegillet**
Medium-level one. There are quite a bit of these "analyze this 2D diagram" problems.

### CODE REVIEW

File: exercises/practice/rectangles/.meta/example.gleam [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+          let position4 = #(x2, y1)
+          let is_rectangle =
+            x1 < x2 && y1 < y2 && map.has_key(corners, position3) && map.has_key(
+              corners,
+              position4,
+            ) && has_vertical_walls_between(map, position1, position4) && has_vertical_walls_between(
+              map,
+              position3,
+              position2,
+            ) && has_horizontal_walls_between(map, position1, position3) && has_horizontal_walls_between(
```
Comment (jiegillet) [inline exercises/practice/rectangles/.meta/example.gleam:61]: There is room to grow for the formatter here :D

Comment (suraj-mandal) [reply exercises/practice/rectangles/.meta/example.gleam:61]: Hi, Can I add some more exercises as per other tracks, verify with the test runner and create a pull request?

Comment (lpil) [reply exercises/practice/rectangles/.meta/example.gleam:61]: Yes of course @suraj-mandal ! That would be great

Comment (lpil) [reply exercises/practice/rectangles/.meta/example.gleam:61]: @jiegillet agreed, there is an issue for this I believe

**Review by lpil — APPROVED** [review]
Nice! Thank you

File: exercises/practice/rectangles/.meta/example.gleam (hunk 1 above)
Comment (jiegillet) [reply exercises/practice/rectangles/.meta/example.gleam:61]: @suraj-mandal make sure you use the generator, it saves tons of time, although it’s not perfect (you are expected to make the test file better manually). Run it from the script in `bin`

Comment (suraj-mandal) [reply exercises/practice/rectangles/.meta/example.gleam:61]: gotcha.. thanks!

**Comment by suraj-mandal** [comment]
Hi @jiegillet. As you suggested I started with an exercise and was using the generate_practice_exercise.sh script to create the exercise, however I am getting a 404 error while accessing this url (https://raw.githubusercontent.com/exercism/gleam-test-runner/main/packages)

It is showing 404 error. Here are the screenshots confirming the same:

image

Are there any extra configuration that are required, or is there some authentication key that is to be provided so that I am able to fetch the required files?

In the meantime I am creating tests and file on my own.

**Comment by suraj-mandal** [comment]
Got it, Adding `185.199.108.133 raw.githubusercontent.com` to hosts file solved the issue

**Comment by suraj-mandal** [comment]
@lpil I

For this line

```
JsonList([head, ..]) -> "List(" <> json_data_to_gleam_type(head) <> ")"
```

I am getting the given error

image

Gleam is not able to consider `<>` as an operator and is assuming < to be a less than symbol. Any way to mitigate this? For this test cases are not passing.

I am using gleam 0.22.1 for this

**Comment by jiegillet** [comment]
@suraj-mandal sorry for he 404 errors, I’ve never seen those before. A potential fix would be to use the version of the repo cloned/cached by configlet, not sure how complex that would be. 

As for the concat issues, you need to update Gleam. The current version is 0.26

**Pull request merged**